src/Components/ProductCard/index.js

```diff
@@ -1,20 +1,71 @@
 /* eslint-disable jsx-a11y/anchor-is-valid */
+import { useState, useEffect } from 'react';
 import classnames from 'classnames';
 
 import moneyFormatter from '../../utilities/moneyFormatter';
 import './style.scss';
 
 const ProductCard = ({ product, className }) => {
+  const [selectedVariant, setSelectedVariant] = useState(false);
+  const [selectedColor, setSelectedColor] = useState(false)
+
   const onClick = () => {
     sessionStorage.productDetails = JSON.stringify(product);
     window.location = `/product-details?product=${product.productID}`;
   }
+
+  const colorMetaName = 'Color';
+  const maxColorOptionsToShow = 5;
+
+  const getColorOptions = () => {
+    const colorOptions = product?.meta_data.filter(md => md.key === colorMetaName);
+    return !!colorOptions && !!colorOptions[0] ? colorOptions[0].values : false;
+  }
+
+  const getSelectableVariantByColorOption = colorOptionValue => {
+    return product?.variants.find(variant => variant.meta_data.find(md => md.key === colorMetaName && md.value === colorOptionValue))
+  }
+
+  const selectColorOption = colorOptionValue => {
+    setSelectedColor(colorOptionValue);
+    if(!!colorOptionValue) {
```

Comment:
@hemant-cbotics can you do this here `if(colorOptionValue) {}`

Reply:
Sure

Reply:
@Nziranziza - updated